AIPscan/API/namespace_report_data.py

```diff
@@ -56,11 +56,68 @@ def get(self, storage_service_id):
         )
 
 
+@api.route("/largest-aips/<storage_service_id>")
+class LargestAIPList(Resource):
+    @api.doc(
+        "list_largest_aips",
+        params={
+            fields.FIELD_START_DATE: {
+                "description": "AIP creation start date (inclusive, YYYY-MM-DD)",
+                "in": "query",
+                "type": "str",
+            },
+            fields.FIELD_END_DATE: {
+                "description": "AIP creation end date (inclusive, YYYY-MM-DD)",
+                "in": "query",
+                "type": "str",
+            },
+            fields.FIELD_LIMIT: {
+                "description": "Number of results to return (default is 20)",
+                "in": "query",
+                "type": "int",
+            },
+            fields.FIELD_STORAGE_LOCATION: {
+                "description": "Storage Location ID",
+                "in": "query",
+                "type": "int",
+            },
+        },
+    )
+    def get(self, storage_service_id, limit=20):
+        """List largest AIPs"""
+        start_date = parse_datetime_bound(request.args.get(fields.FIELD_START_DATE))
+        end_date = parse_datetime_bound(
+            request.args.get(fields.FIELD_END_DATE), upper=True
+        )
+        storage_location_id = request.args.get(fields.FIELD_STORAGE_LOCATION)
+        try:
+            limit = int(request.args.get(fields.FIELD_LIMIT, 20))
+        except ValueError:
+            pass
```

Comment:
Should limit be set to 20 if their is a ValueError?

Reply:
yes, good point!